src/main/java/com/greenfox/tribes1/ApplicationUser/ApplicationUserController.java

```diff
@@ -15,14 +15,14 @@
 public class ApplicationUserController {
   
   private ApplicationUserService applicationUserService;
-  
+
   @Autowired
   public ApplicationUserController(ApplicationUserService applicationUserService) {
     this.applicationUserService = applicationUserService;
   }
   
   @PostMapping("/register")
-  public ResponseEntity register(@Valid @RequestBody ApplicationUserDTO applicationUserDTO) throws UsernameTakenException, NotValidKingdomNameException {
+  public ResponseEntity register(@Valid @RequestBody ApplicationUserDTO applicationUserDTO) throws UsernameTakenException, NotValidKingdomNameException, TroopIdNotFoundException, BuildingNotValidException, NotValidTypeException, TroopNotValidException, BuildingIdNotFoundException {
```

Comment:
What about creating a superclass for them? For example you'd create sth. like RegisterException and inherit all of these from that class, thus you wouldn't have to throw all of these Expressions, just a RegisterExpression

Reply:
+1 :)